src/Neo.VM/Types/CompoundType.cs

```diff
@@ -60,12 +60,32 @@ public sealed override bool GetBoolean()
         }
 
         /// <summary>
-        /// The operation is not supported. Always throw <see cref="NotSupportedException"/>.
+        ///
+        /// This method provides a hash code for the <see cref="CompoundType"/> based on its item's span.
+        /// It is used for efficient storage and retrieval in hash-based collections.
+        ///
+        /// Use this method when you need a hash code for a <see cref="CompoundType"/>.
         /// </summary>
-        /// <exception cref="NotSupportedException">This method always throws the exception.</exception>
+        /// <returns>The hash code for the <see cref="CompoundType"/>.</returns>
         public override int GetHashCode()
         {
-            throw new NotSupportedException();
+            var h = new HashCode();
+            h.Add(Count);
+            h.Add(Type);
+            foreach (var item in SubItems)
+            {
+                // This isn't prefect and leaves somethings unsolved.
+                if (item is CompoundType cItem)
```

Comment:
There is such a cause here:
```
var map1 = new Map();
            map1[1] = 2;
            map1[3] = 4;

            var map2 = new Map();
            map2[3] = 4;
            map2[1] = 2;
        
            Assert.AreEqual(map1.GetHashCode(), map2.GetHashCode());

```

`Assert.AreEqual` failed.
Because `VM.Map` uses `OrderedDictionary` internally, but `OrderedDictionary` is not ordered.

Reply:
It could be fixed in a new PR